packages/appstore/src/components/main-title-bar/__tests__/main-title-bar.spec.tsx

```diff
@@ -0,0 +1,206 @@
+        const eu_btn = screen.queryByText('EU');
+        const non_eu_btn = screen.queryByText('Non-EU');
+        const ic_icon_info_outline = screen.queryByTestId('dt-ic-info-outline');
+
+        expect(container).toBeInTheDocument();
+        expect(regulator_switcher_mobile).toBeInTheDocument();
+        expect(eu_btn).toBeInTheDocument();
+        expect(non_eu_btn).toBeInTheDocument();
+        expect(ic_icon_info_outline).toBeInTheDocument();
+    });
+
+    it('should render the the <RegulatorSwitcherLoader/> when "is_switching" is true on mobile', () => {
+        const history = createBrowserHistory();
+        const mock = mockStore({
+            client: {
+                is_landing_company_loaded: true,
+                is_switching: true,
+            },
+            traders_hub: {
+                content_flag: 'low_risk_cr_non_eu',
+            },
+        });
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <Router history={history}>
+                <StoreProvider store={mock}>{children}</StoreProvider>
+            </Router>
+        );
+
+        const { container } = render(<MainTitleBar />, { wrapper });
+        const regulator_switcher_mobile = screen.queryByTestId('dt-regulators-switcher-mobile');
+        const regulator_switcher_mobile_loader = screen.queryByTestId('dt-regulators-switcher-loading-mobile');
+        const eu_btn = screen.queryByText('EU');
+        const non_eu_btn = screen.queryByText('Non-EU');
+        const ic_icon_info_outline = screen.queryByTestId('dt-ic-info-outline');
+
+        expect(container).toBeInTheDocument();
+        expect(regulator_switcher_mobile).toBeInTheDocument();
+        expect(regulator_switcher_mobile_loader).toBeInTheDocument();
+        expect(eu_btn).not.toBeInTheDocument();
+        expect(non_eu_btn).not.toBeInTheDocument();
+        expect(ic_icon_info_outline).not.toBeInTheDocument();
+    });
+
+    it('should execute function when clicked on one of the tabs on the RegulatorSwitcher tab on mobile', () => {
+        const history = createBrowserHistory();
+        const mock = mockStore({
+            client: {
+                is_landing_company_loaded: true,
+            },
+            traders_hub: {
+                content_flag: 'low_risk_cr_non_eu',
+            },
+        });
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <Router history={history}>
+                <StoreProvider store={mock}>{children}</StoreProvider>
+            </Router>
+        );
+
+        const { container } = render(<MainTitleBar />, { wrapper });
+        const regulator_switcher_mobile = screen.queryByTestId('dt-regulators-switcher-mobile');
+        const eu_btn = screen.getByText('EU');
+        const non_eu_btn = screen.getByText('Non-EU');
+        const ic_icon_info_outline = screen.queryByTestId('dt-ic-info-outline');
+
```

Comment:
@aizad-deriv do we again need to check whether the entire component has rendered correctly before testing the triggering of the function? 🤔 We are already testing that in the tests above for both desktop/mobile.
Please correct me if I am wrong 🙏

Reply:
@aum-deriv I mean it's not wrong but you make a very good point. Should I remove it? Let's see what the other's say

cc: @farzin-deriv @thisyahlen-deriv @suisin-deriv